owncloudApp/src/main/java/com/owncloud/android/presentation/ui/settings/fragments/SettingsFragment.kt

```diff
@@ -0,0 +1,227 @@
+                    enablePatternLauncher.launch(intent)
+                } else {
+                    intent.action = PatternLockActivity.ACTION_CHECK_WITH_RESULT
+                    disablePatternLauncher.launch(intent)
+                }
+            }
+            false
+        }
+
+        // Biometric lock
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
+            prefSecurityCategory?.removePreference(prefBiometric)
+        } else if (prefBiometric != null) {
+            // Disable biometric lock if Passcode or Pattern locks are disabled
+            if (prefPasscode?.isChecked == false && prefPattern?.isChecked == false) {
+                disableBiometric(getString(R.string.prefs_biometric_summary))
+            }
+            prefBiometric?.setOnPreferenceChangeListener { preference: Preference?, newValue: Any ->
+                val incomingValue = newValue as Boolean
+
+                // Biometric not supported
+                if (incomingValue && biometricManager?.isHardwareDetected == false) {
+                    showMessageInSnackbar(getString(R.string.biometric_not_hardware_detected))
+                    return@setOnPreferenceChangeListener false
+                }
+
+                // No biometric enrolled yet
+                if (incomingValue && biometricManager?.hasEnrolledBiometric() == false) {
+                    showMessageInSnackbar(getString(R.string.biometric_not_enrolled))
+                    return@setOnPreferenceChangeListener false
+                }
+                true
+            }
+        }
+
+        // Touches with other visible windows
+        prefTouchesWithOtherVisibleWindows?.setOnPreferenceChangeListener { preference: Preference?, newValue: Any ->
+            if (newValue as Boolean) {
+                activity?.let {
+                    AlertDialog.Builder(it)
+                        .setTitle(getString(R.string.confirmation_touches_with_other_windows_title))
+                        .setMessage(getString(R.string.confirmation_touches_with_other_windows_message))
+                        .setNegativeButton(getString(R.string.common_no), null)
+                        .setPositiveButton(
+                            getString(R.string.common_yes)
+                        ) { dialog: DialogInterface?, which: Int ->
+                            settingsViewModel.setPrefTouchesWithOtherVisibleWindows(true)
+                            prefTouchesWithOtherVisibleWindows?.isChecked = true
+                        }
+                        .show()
+                }
+                return@setOnPreferenceChangeListener false
+            }
+            true
+        }
+
+    }
+
+    private fun enableBiometric() {
+        prefBiometric?.isEnabled = true
+        prefBiometric?.summary = null
+    }
+
+    private fun disableBiometric(summary: String) {
+        if (prefBiometric?.isChecked == true) {
+            prefBiometric?.isChecked = false
+        }
```

Comment:
What if you remove this check?
```suggestion
            prefBiometric?.isChecked = false
```

Reply:
Not worth it to check if the preference is checked everytime you want to disable it, I agree